src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/RazorDocumentServiceBase.cs

```diff
@@ -24,31 +25,57 @@ internal abstract class RazorDocumentServiceBase(in ServiceArgs args) : RazorBro
     protected virtual IDocumentPositionInfoStrategy DocumentPositionInfoStrategy { get; } = DefaultDocumentPositionInfoStrategy.Instance;
 
     protected DocumentPositionInfo GetPositionInfo(RazorCodeDocument codeDocument, int hostDocumentIndex)
+        => GetPositionInfo(codeDocument, hostDocumentIndex, preferCSharpOverHtml: false);
+
+    protected DocumentPositionInfo GetPositionInfo(RazorCodeDocument codeDocument, int hostDocumentIndex, bool preferCSharpOverHtml)
     {
-        return DocumentPositionInfoStrategy.GetPositionInfo(DocumentMappingService, codeDocument, hostDocumentIndex);
+        var positionInfo = DocumentPositionInfoStrategy.GetPositionInfo(DocumentMappingService, codeDocument, hostDocumentIndex);
+
+        if (preferCSharpOverHtml && positionInfo.LanguageKind == RazorLanguageKind.Html)
+        {
+            // Sometimes Html can actually be mapped to C#, like for example component attributes, which map to
+            // C# properties, even though they appear entirely in a Html context. Since remapping is pretty cheap
+            // it's easier to just try mapping, and see what happens, rather than checking for specific syntax nodes.
+            if (DocumentMappingService.TryMapToGeneratedDocumentPosition(codeDocument.GetCSharpDocument(), positionInfo.HostDocumentIndex, out VsPosition? csharpPosition, out _))
+            {
+                // We're just gonna pretend this mapped perfectly normally onto C#. Moving this logic to the actual position info
+                // calculating code is possible, but could have untold effects, so opt-in is better (for now?)
+
+                // TODO: Not using a with operator here because it doesn't work in OOP for some reason.
```

Comment:
Really? Was it not working in `RemoteGoToDefinitionService` where it was used previously? I don't recall seeing it fail there, but maybe I wasn't looking carefully?

Reply:
No idea what is going on, but it's a 100% repro for me. It even repros in my regular VS main build, without the exp hive or any locally built bits. Turn on cohosting, open up a Razor file, press F12 on an attribute, get a yellow bar.

I checked DLL versions etc. and everything looked up to date.

Reply:
Interesting. Would you mind filing an issue so there's more of record than a TODO comment?

Reply:
https://github.com/dotnet/razor/issues/10843